CMakeLists.txt

```diff
@@ -36,6 +37,23 @@ if (MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -wd4125")
 endif()
 
+# Enable ccache
+# LLVM_CCACHE_BUILD for consistency, but this currently has no effect
+set(LLVM_CCACHE_BUILD ${ONNX_MLIR_CCACHE_BUILD})
+if(ONNX_MLIR_CCACHE_BUILD)
+  find_program(CCACHE_PROGRAM ccache)
+  if(CCACHE_PROGRAM)
+    if(CMAKE_SYSTEM_NAME MATCHES "Windows")
+      set(CMAKE_C_COMPILER_LAUNCHER ${CCACHE_PROGRAM})
+      set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_PROGRAM})     
+    else()
+      set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${CCACHE_PROGRAM})
```

Comment:
This would technically not work for Xcode which is used on a non-Windows platform

Reply:
I don't have a Mac to try, but some quick searching shows that ccache with ninja and Xcode requires some [non-trivial additional support](https://stackoverflow.com/a/36515503) anyway, which seems out of scope here.

Reply:
Can we add a comment here saying so?